src/Database/Table/GroupedSelection.php

```diff
@@ -201,6 +201,10 @@ protected function loadRefCache()
 		}
 	}
 
+	protected function emptyResultSet($saveCache = TRUE, $deleteRererencedCache = TRUE) {
+		parent::emptyResultSet($saveCache, false);
```

Comment:
running line `$this->refCache['referenced'] = [];` for GroupedSelection was actually the problem that i am trying to solve. This will actually break the `query.count.phpt` test